packages/flutter_test/lib/src/window.dart

```diff
@@ -610,16 +619,36 @@ class FakeAccessibilityFeatures implements ui.AccessibilityFeatures {
 ///     testing and mocking purposes.
 class TestPlatformDispatcher implements ui.PlatformDispatcher {
   /// Constructs a [TestPlatformDispatcher] that defers all behavior to the given
-  /// [dart:ui.PlatformDispatcher] unless explicitly overridden for test purposes.
+  /// [ui.PlatformDispatcher] unless explicitly overridden for test purposes.
   TestPlatformDispatcher({
     required ui.PlatformDispatcher platformDispatcher,
-  }) : _platformDispatcher = platformDispatcher;
+  }) : _platformDispatcher = platformDispatcher {
+    _configuration = _platformDispatcher.configuration.copyWith();
+    _updateViews();
+  }
 
-  /// The [dart:ui.PlatformDispatcher] that is wrapped by this [TestPlatformDispatcher].
+  /// The [ui.PlatformDispatcher] that is wrapped by this [TestPlatformDispatcher].
   final ui.PlatformDispatcher _platformDispatcher;
 
   @override
-  ui.VoidCallback? get onMetricsChanged => _platformDispatcher.onMetricsChanged;
+  TestFlutterView? get implicitView {
+    return _platformDispatcher.implicitView != null ?
+      _testViews[_platformDispatcher.implicitView!.viewId]! :
+      null;
+  }
+
+  @override
+  ui.PlatformConfiguration get configuration => _configuration;
+  late ui.PlatformConfiguration _configuration;
+
+  final Map<Object, TestFlutterView> _testViews = <Object, TestFlutterView>{};
+
+  @override
+  ui.VoidCallback? get onMetricsChanged => () {
```

Comment:
It's odd that the getter doesn't return what was set by the setter.

Reply:
Also, don't you need to somehow register your own listener to the real platformdispatcher that will do the _updateViews thing when it fires and then forward to the one registered on the fake?

Reply:
For all intents and purposes, this _is_ the real `PlatformDispatcher` when tests are running.

Since it's just a callback, I can't set it to multiple different values. Because of this, I couldn't just set it to be `_updateViews` as the user or system setting the callback would override mine. I also couldn't just have it always _return_ `_updateViews`, because that'd keep the system and the users from being able to add a callback here.

To allow `_updateViews` to get called on each `onMetricsChanged` call _and_ let the user set some custom value as well, I saw two options:

1. Override the getter to call `_updateViews` and then whatever has been registered by the system or users
2. Override the setter to always set the value to `_updateViews` and whatever value is passed in. Then in the constructor, call the setter with null or a bogus callback to have it initialize with `_updateViews`.

I went with the former because it's simpler. If you have another way to register both my `_updateViews` _and_ whatever the user may want to set, l'd be happy to hear it and see if it works.

Reply:
I would have expected something like this:

```dart
class TestPlatformDispatcher {
  TestPlatformDispatcher({PlatformDispatcher pd}) : _pd = pd {
   assert(_pd.onMetricsChanged == null);
    _pd.onMetricsChanged = _handleMetricsChanged;
  }

 final PlatformDispatcher _pd;

  @override
  VoidCallback? get onMentricsChanged => _onMetricsChanged;
  VoidCallback? _onMetricsChanged;
  @override
  set onMentricsChanged(VoidCallback value) {
    _onMetricsChanged = value;
  }

 void _handleMetricsChanged() {
    _updateViews();
   _onMetricsChanged?.call();
  }
}
```

In your current implementation I don't understand how _updateViews is executed when the platform adds a new view and the real PlatformDispatcher fires onMetricsChanged. Maybe I miss something?

Another option would be to do _updateView lazily whenever someone accesses TestPlatformDispatcher.views.

Reply:
Nope, it was me missing something here. I assumed that things external to `onMetricsChanged` would be the ones calling the method when they updated `PlatformDispatcher`.

After you mentioned it here though, you're mostly right. The one thing I'll change from your suggestion is that the `onMetricsChanged` override will return the `_platformDispatcher.onMetricsChanged` directly so that if someone _does_ call it from the binding, it still gets everything, not just the user supplied override.

Thanks for clarifying for me!